src/pocketmine/plugin/PluginManager.php

```diff
@@ -269,6 +269,16 @@ public function loadPlugins(string $directory, array $newLoaders = null){
 						continue;
 					}
 
+					$pluginPhpVersions = $description->getCompatiblePhpVersions();
+					$pluginCompatiblePhpVersions = array_filter($pluginPhpVersions, function(string $version) : bool{ return Utils::arePhpVersionsCompatible(PHP_MAJOR_VERSION . "." . PHP_MINOR_VERSION . "." . PHP_RELEASE_VERSION, $version); });
```

Comment:
I think `version_compare()` should be sufficient, given that it was designed to handle PHP's own versions.